File: tttracker/src/main/java/ch/rasc/tttracker/domain/Company.java
package ch.rasc.tttracker.domain;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import org.springframework.data.jpa.domain.AbstractPersistable;

import ch.rasc.extclassgenerator.Model;
import ch.rasc.extclassgenerator.ModelField;
import ch.rasc.extclassgenerator.ModelType;

@Entity
@Model(value = "TTT.model.Company")
@ModelField(value = "id", type = ModelType.INTEGER)
public class Company extends AbstractPersistable<Long> {

	private static final long serialVersionUID = 1L;

	@NotNull
	@Size(min = 1, max = 200)
	private String name;

	@OneToMany(cascade = CascadeType.ALL, mappedBy = "company")
	private List<Project> projects;

	public String getName() {
		return this.name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public List<Project> getProjects() {
		return this.projects;
	}

	public void setProjects(List<Project> projects) {
		this.projects = projects;
	}

}

File: tttracker/src/main/webapp/app/model/Company.js
Ext.define("TTT.model.Company",
{
  extend : "Ext.data.Model",
  fields : [ {
    name : "id",
    type : "int"
  }, {
    name : "name",
    type : "string"
  } ],
  validations : [ {
    type : "presence",
    field : "name"
  }, {
    type : "length",
    field : "name",
    min : 1,
    max : 200
  } ]
});